artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/codec/AddressStatusEncoding.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.persistence.impl.journal.codec;
+
+import org.apache.activemq.artemis.api.core.ActiveMQBuffer;
+import org.apache.activemq.artemis.core.journal.EncodingSupport;
+import org.apache.activemq.artemis.core.persistence.AddressStatus;
+import org.apache.activemq.artemis.utils.DataConstants;
+
+public class AddressStatusEncoding implements EncodingSupport {
+
+   private AddressStatus status;
+
+   private long addressId;
+
+   private long id;
+
+   public AddressStatusEncoding(long addressId, AddressStatus status) {
+      this.status = status;
+      this.addressId = id;
```

Comment:
This assignment seems incorrect. It should almost certainly be `this.addressId = addressId;`.